fintrack/src/context/AuthContext.js

```diff
@@ -0,0 +1,35 @@
+import { createContext, useState, useEffect } from 'react'
+import Cookies from 'js-cookie'
+
+export const AuthContext = createContext(null)
+
+export const AuthProvider = ({ children }) => {
+
+    const [authToken, setAuthToken] = useState(null)
```

Comment:
Also, this complexity throughout the `authToken` being `null` could be alleviated if you simply blocked the render on this call to `Cookies.get` and waited 0.00000001 seconds for that function call to complete, and _then_ rendered your whole app. That way, you would never be getting `null` from `useContext(AuthContext)`!